tests/run-make/symbol-mangling-hashed/rmake.rs

```diff
@@ -0,0 +1,130 @@
+
+        let dynamic_symbols = exported_dynamic_symbol_names(&dylib_file);
+
+        if dynamic_symbols.iter().filter(|sym| sym.contains("hello")).count() != 0 {
+            eprintln!("exported dynamic symbols: {:#?}", dynamic_symbols);
+            panic!("expected no occurrence of `hello`");
+        }
+
+        let expected_prefix =
+            if is_darwin() { "__RNxC12hashed_dylib" } else { "_RNxC12hashed_dylib" };
+        if dynamic_symbols.iter().filter(|sym| sym.starts_with(expected_prefix)).count() != 2 {
+            eprintln!("exported dynamic symbols: {:#?}", dynamic_symbols);
+            panic!("expected two dynamic symbols starting with `{expected_prefix}`");
+        }
+    }
+
+    {
+        let dylib_filename = dynamic_lib_name("default_dylib");
+        println!("checking so `{dylib_filename}`");
+
+        let dylib_blob = rfs::read(&dylib_filename);
+        let dylib_file = object::File::parse(&*dylib_blob)
+            .unwrap_or_else(|e| panic!("failed to parse `{dylib_filename}`: {e}"));
+
+        let dynamic_symbols = exported_dynamic_symbol_names(&dylib_file);
+
+        if dynamic_symbols
+            .iter()
+            .filter(|sym| sym.contains("default_dylib") && sym.contains("hello"))
+            .count()
+            != 1
+        {
+            eprintln!("exported dynamic symbols: {:#?}", dynamic_symbols);
+            panic!("expected one occurrence of mangled `hello`");
+        }
+
+        let expected_rlib_prefix =
+            if is_darwin() { "__RNxC11hashed_rlib" } else { "_RNxC11hashed_rlib" };
+        if dynamic_symbols.iter().filter(|sym| sym.starts_with(expected_rlib_prefix)).count() != 2 {
+            eprintln!("exported dynamic symbols: {:#?}", dynamic_symbols);
+            panic!("expected two exported symbols starting with `{expected_rlib_prefix}`");
+        }
+
+        let expected_dylib_prefix =
+            if is_darwin() { "__RNxC12hashed_dylib" } else { "_RNxC12hashed_dylib" };
+        if dynamic_symbols.iter().any(|sym| sym.starts_with("_RNxC12hashed_dylib")) {
+            eprintln!("exported dynamic symbols: {:#?}", dynamic_symbols);
+            panic!("did not expect any symbols starting with `{expected_dylib_prefix}`");
+        }
+    }
+
+    {
+        let bin_filename = bin_name("default_bin");
+        println!("checking bin `{bin_filename}`");
+
+        let bin_blob = rfs::read(&bin_filename);
+        let bin_file = object::File::parse(&*bin_blob)
+            .unwrap_or_else(|e| panic!("failed to parse `{bin_filename}`: {e}"));
+
+        let dynamic_symbols = exported_dynamic_symbol_names(&bin_file);
+
+        let expected_rlib_prefix =
+            if is_darwin() { "__RNxC11hashed_rlib" } else { "_RNxC11hashed_rlib" };
+        let expected_dylib_prefix =
+            if is_darwin() { "__RNxC12hashed_dylib" } else { "_RNxC12hashed_dylib" };
+        if dynamic_symbols.iter().any(|sym| {
```

Comment:
this just checks that a binary doesn't export any symbols, which isn't a very useful thing to test for this test.
I think we should just drop this binary part.

Reply:
Dropped symbol checks on the bin